### ISSUE TITLE
Overloaded functions cause `reportIncompatibleMethodOverride` during multiple inheritance

### ISSUE THREAD

**Issue by neerajanil** [microsoft/pyright#9625]
**Describe the bug**
During multiple inheritance if both base classes implement the same method with the same override signatures, it will still throw a `reportIncompatibleMethodOverride` error on the child class. `mypy` does not throw an error for the same code.
[pyright playground](https://pyright-play.net/?code=GYJw9gtgBALgngBwJYDsDmUkQWEMpgBuApiADZgCGAJgDRQCqKSYK9AMkgM4wCwAUAIDGZSly5QAygFcEpAFwCoyqEpUABIqQo01y6sWBQIxGAAsw1ABRdiZYPQAe8zChgBKKAFoAfK5iK-CrBUAB04XohkZok5FTUkQZGJuaWNnYOUM5QPCCevjkwIIEhKuGhkcGJhsamFta29k4uTCwoANqoMPS5ALr5fq2snW49Rb0lpVAgptIgKFmRwqLiUtIARgCMVjJyeZPKMdrx1cl1aY2Z2V0D-gch5ZUq0VpxukEqSbWpDRnNhXlvH5cvdgo8PqVTt96ukmlkWsxhl0xiB%2BkDGIiOsiARMnsoZjA5gtHAJlmIJDJ1gAmHayUjue5HN4JCFQL4pGGXf43dFdUFlCKsqqspk6FnBdnnX5w7K5W4gvFhQVTVSsyU-WFXBFtEbdHG3IZY0Y4-n42bzRaCK0iclrdYAZlpW3olKpDKgAGJpsQcHgAJIoISQBCUGBIdZkYgAWSlAHlYiAkAZvb6YMRqFAzKRiJFykA)
[mypy playground](https://mypy-play.net/?mypy=latest&python=3.10&gist=2d075f34355da0871425050cea1134e8)

**Code**
```python
from typing import overload, Union, List

class Super:
    
    @overload
    def method(self, x: int) -> int:
        ...
        
    @overload
    def method(self, x: str) -> str:
        ...
        
    def method(self, x: Union[int, str]) -> Union[int, str]:
        return x
    
class Sub1(Super):
    @overload
    def method(self, x: int) -> int:
        ...
        
    @overload
    def method(self, x: str) -> str:
        ...
        
    def method(self, x: Union[int, str]) -> Union[int, str]:
        return x

class Sub2(Super):
    @overload
    def method(self, x: int) -> int:
        ...
        
    @overload
    def method(self, x: str) -> str:
        ...
        
    def method(self, x: Union[int, str]) -> Union[int, str]:
        return x


class Sub3(Sub1, Sub2): # reportIncompatibleMethodOverride reported here
    ...
```

**VS Code extension or command-line**
Running pyright CLI version 1.1.391

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.392.
